### PR TITLE
Fix empty slug leaving `hasUnpersistedChanges()` set to true

### PR DESCRIPTION
**Pull request by attitude**
### Checks
- [x] The changes and implementation strategy have been consulted with the maintainers.
- [x] The code passes all the checks (Run `pnpm run check`).

### CODE REVIEW

File: packages/binding/src/entityEvents/useDerivedField.ts [hunk 1]
```diff
@@ -48,8 +48,7 @@ export const useDerivedField = <SourceValue extends FieldValue = FieldValue>(
 					return
 				}
 
-				const transformedValue = transform(sourceAccessor.value)
-				derivedAccessor.updateValue(transformedValue, {
+				derivedAccessor.updateValue(sourceAccessor.value === null ? null : transform(sourceAccessor.value), {
```
Comment (VojtaStanek) [inline packages/binding/src/entityEvents/useDerivedField.ts:51]: This is quite a change in behaviour. 
a) If you want go this route, change the type on line 20 in this file.
b) But I would prefer to do it in `SlugField` - handle there the case when driver field is null and return null in that case.

**Review by VojtaStanek — CHANGES_REQUESTED** [review]

File: packages/binding/src/entityEvents/useDerivedField.ts (hunk 1 above)
Comment (attitude) [reply packages/binding/src/entityEvents/useDerivedField.ts:51]: @matej21 can you respond, since you proposed this change?

Comment (attitude) [reply packages/binding/src/entityEvents/useDerivedField.ts:51]: IMHO, when we have schema exposed on the field done in #168, we could also check `nullable: true` in the parser.

1. Should parser return `null` only when `null` is allowed?
1. Should parser return `null` when the value on server is still `null` even though `nullable: false`?
1. What should be returned when parsing was not possible?

Which of these should parser have resposibility to implement?

Comment (matej21) [reply packages/binding/src/entityEvents/useDerivedField.ts:51]: Yep, doing this in slugfield is a also possible (I've proposed this alternative approach in that slack message). I'm ok with both solutions.

**Review by VojtaStanek — APPROVED** [review]

**Pull request merged**